tests/naive.py

```diff
@@ -6,7 +6,7 @@
 from stumpy import core, config
 
 
-def z_norm(a, axis=0, threshold=1e-7):
+def z_norm(a, axis=0, threshold=config.STUMPY_STDDEV_THRESHOLD):
```

Comment:
I think we should move this into the function as:

```
std[np.less(std, config.STUMPY_STDDEV_THRESHOLD, where=~np.isnan(std))] = 1.0
```

Right now, no tests actually use/alter the `threshold`. Additionally, I am wondering if we are hiding some other precision issues by doing this? Why do we need to change this at all? What is the test that would be failing?

Reply:
> Why do we need to change this at all?

I think (one of) the reason(s) is that `naive function stump` calls function `distance_profile`, where `z_norm` is used. 
```
def distance_profile(Q, T, m)
     ## some preprocessing ##
     D = np.linalg.norm(z_norm(core.rolling_window(T, m), 1) - z_norm(Q), axis=1)
     return D
```



And, if we take a look at the performant version of stump, we can see it calls function ` core.preprocess_diagonal`, which contains the following lines of code:
```
T_subseq_isconstant = Σ_T < config.STUMPY_STDDEV_THRESHOLD
Σ_T[T_subseq_isconstant] = 1.0  # Avoid divide by zero in next inversion step
```

So, I took a look at the value of `config.STUMPY_STDDEV_THRESHOLD` and I noticed it is set to 1e-7, the same value used in naive version of z_norm. And, since both cases have the same purpose for using 1e-7, I decided to replace the hard-coded value 1e-7 in `naive z_norm` with a  config variable.

Reply:
@seanlaw 
What is the test that would be failing?

I think no test is failing because currently `config.STUMPY_STDDEV_THRESHOLD` is 1e-7, which is the same hard-coded value used in `naive z_norm`. So, I thought it would make sense to use config variable here.

And, then I realized that I need to change the `performant z_norm` as well since the threshold there is 0.0  (not 1e-7).

Reply:
@seanlaw 

* This whole thing is not related to this PR! :)  Maybe I should just leave it. It is more related to the `fix_precision` PR. 
* Also, maybe we should just use `threshold 0.0` in the `naive z_norm` and `performant z_norm` (and forget about everything I said) and see if we get an error or not.

Reply:
Yeah, maybe let's do it elsewhere. 

Having a threshold of 0.0 seems weird as it implies that the values can be negative